### ISSUE TITLE
SurvivalGames Arenas not ending properly 

### ISSUE THREAD

**Issue by ghost** [Double0negative/Survival-Games#95]
I run a massive Server and i am using the survivalgames plugin. When my arenas end, When you click to sign it says 'Cant join arena' the if you look at the sign it says 'FINISHING' and it only fixes if i restart the server but happens again when it ends.
I have redownloaded the plugin and tried a different version and this is still occurring. It does come with a crash when this happens. Please help me out!
Here is the crash http://pastebin.com/YSrD9jRV
Thanks A Lot

**Comment by Pocketkid2** [comment]
I will take a look. There are a lot of bugs like this though.

**Comment by Pocketkid2** [comment]
Is there any more error messages?

**Comment by Pocketkid2** [comment]
https://github.com/Double0negative/Survival-Games/pull/96
This is what I think will fix the problem

**Comment by ghost** [comment]
So it's something to do with spectating that is causing it to happen?

**Comment by ghost** [comment]
And there are no other crashes that come with it sorry

**Comment by Pocketkid2** [comment]
Well, if you read the stack trace, it says it's getting a NullPointerException in the code for removing spectators. I took a look at the code, and I did some fixing up that might fix it. This whole plugin is full of crappy code. I'm thinking about making my own stable version

**Comment by Pocketkid2** [comment]
I am with you though I have noticed this problem since I started using it practically :)

**Comment by ghost** [comment]
Yeah it is crap code. That would be awesome for you to make your own. I would even pay! Do just to be clear, what should I do to fix this bug so my arenas stay working ?

**Comment by Pocketkid2** [comment]
Well, if the author approves my pull request, then maybe the bug is fixed. I will do some extensive testing, and coding of my own. I can get back to you if I get a build that works.

**Comment by ghost** [comment]
Oh please do! Thanks so much for that :) if you would like add my skype chrisbadaoui2000

**Comment by Pocketkid2** [comment]
Ok. I really enjoy coding Bukkit plugins. I mostly do it just for the fun of making something, and fixing things that nobody has time to fix. it's sorta my hobby :)

**Comment by Pocketkid2** [comment]
This is the first build of my in-progress re-creating of SurvivalGames. Check it out, and report any errors. We can set up skype, but later. :) http://www.mediafire.com/download/q72wd6k8dpc5ddc/SurvivalGamesReloaded.jar

**Comment by Pocketkid2** [comment]
BTW It is expected to have LOTS of errors.

**Comment by Pocketkid2** [comment]
Just fixed some more initial transfer bugs. Here ya go: http://www.mediafire.com/?hh0ysji30dqja7z

**Comment by ghost** [comment]
Do I set it up like the original plugin?

**Comment by Pocketkid2** [comment]
Sure. Beware it should have bugs, so you might want to just make sure your safely save and restart your server

**Comment by Pocketkid2** [comment]
I can't acces skype right now, but I will contact you via skype ASIC

**Comment by ghost** [comment]
No problem! Once you add me on skype I will report all the bugs to you :)

**Comment by Pocketkid2** [comment]
Ok... But you can start reporting any first bugs right now if you would. (I would like that)

**Comment by ghost** [comment]
No problem. Did you fix the bug where you can join arenas though ?

**Comment by ghost** [comment]
Also is the plugin compatible with 1.5.2? As my server is 1.5.2

**Comment by ghost** [comment]
With the SG plugin you made, i wanted to ask are you able to fix deathmatch up? So it counts from 10-0 then gets teleported to the middle? And do you recon its hard to add statistics to the side? Like when you join the arena, it tells you how many players have died, alive, and what arena number? Thanks so much keep up the awesome work! 
BTW was this plugin meant to fix the 'cant join arena' problem?

**Comment by ghost** [comment]
Here is my first bug, when i try to join the arena it says ' joining arena 1' then i get tped to where i set the spawn points but i can move around and the signs dont say that i joined the arena. Here is also the error 
http://pastebin.com/J42ZKbzv

**Comment by Pocketkid2** [comment]
Alright. This is just a test, to see if I can get the code to work in a new repo controlled by me, It should work with older builds, though I do compile for 1.6.4 R2.0. After I fix all the initial bugs I should be able to finish implementing all the things the previous author forgot (sponsor, deathmatch, etc...). For now I will focus on the main bugs.

**Comment by Pocketkid2** [comment]
LAWL You're error was a syntax error on my part :)

**Comment by Pocketkid2** [comment]
Alright for now try this: http://www.mediafire.com/?7zdtufe5bqnd26d

**Comment by ghost** [comment]
Thanks

**Comment by ghost** [comment]
I removed spectating and now my arenas are fine ! :D Does your build allow spectators and has that bug fixed?

**Comment by Pocketkid2** [comment]
What do you mean you 'removed spectating'? I might have fixed the bug, why don't you check for me?

**Comment by PatchGuru** [comment]
Sounds to me like your bug is in the code, that when a match ends, the plugin runs through its protocols, either missing or skipping the step that would remove the spectators from the arena once it stops, leaving it stuck in the "FINISHING" mode. I suggest you remove the entire spectating feature of this plugin or fix this bug, I gave up on this plugin when @Double0negative stopped authoring/merging solid pull requests. @Double0negative if you care about this plugin and its future, hand it off to a more capable programmer or get to work merging these solid pull requests.

**Comment by ghost** [comment]
Thanks so much! I removed spectators and now its working :+1:

**Comment by PatchGuru** [comment]
I'm glad to hear that, check my issue request I made regarding all issues.

**Comment by ghost** [comment]
Pocketkid2, Sending me the source code for the latest 1.5.2 survivalgames plugin

**Comment by ghost** [comment]
Can you send me the source code for the latest 1.5.2 survivalgames plugin ****\* SORRY

**Comment by Pocketkid2** [comment]
Why do you need the source code? I don't have 1.5.2 source, I just have my version that i'm working on for 1.6.4 RB. By the way, I got sponsoring to work!

**Comment by ghost** [comment]
Oh, a friend send he can fix a bug i am having with it but he said he needs the source code :/ I already asked you but you said you dont know what it is and it is related 1.5.2 but its the actual plugin itself

**Comment by Pocketkid2** [comment]
This IS the source right here! This github repo is all the current source for the plugin. I have no idea why he would want 1.5.2 as that is kinda old... (Sorry, I don't really understand what you are trying to say...)

**Comment by VillagerJeff** [comment]
Most of the code. I have a version of Survival-Games (but not the source code) where the chest levels working. The source code in the repository appears to be missing a file named "test". We don't care because we prefer to use one chest level.
